Windows fixes (VIAME)

neal-siekierski opened the pull request:
This PR includes fixes to the `viame/master` branch for it to compile on Windows, specifically with Visual Studio 15. Perhaps some of this is also relevant for `master`.

These are largely straightforward changes, but I have a couple concerns:
- What's the canonical place in a `CMakeLists.txt` file to call `add_compile_options`?
- `KipQueryService` requires C++17 because of one use of `[[maybe_unused]]`. Is it worth bumping up the required standard for this?

@mwoehlke-kitware @mattdawkins

mwoehlke-kitware commented on CMakeLists.txt:
```diff
@@ -4,6 +4,13 @@ set(CMAKE_CXX_STANDARD 11)
 project(visGUI)
 
+if(MSVC AND (MSVC_VERSION GREATER_EQUAL 1914))
+  # qtExtensions uses the __cplusplus macro to test for C++11 support
```
This should be fixed by kitware/qtextensions@a1ebde053ccf71f60f1e6928ec04fea1aa025a28; are you sure you need this?

neal-siekierski replied:
The version of qtextensions I'm using has that commit. In building VIAME, however, it doesn't seem to use that flag when building vivia.

mwoehlke-kitware replied:
That's... concerning. Does your `qtExtensionsConfigTargets.cmake` have exported compile options for `qtExtensionsHeaders`? (AFAIK it is working for other people...)

neal-siekierski replied:
It does not. Perhaps the problem is here, with the usage of `ADDED_FLAG`?:
https://github.com/Kitware/qtextensions/blob/a911d919bc8f01c6f3af067dc95a654b2f27cf16/cmake/config.cmake#L88

That variable is set using `PARENT_SCOPE`, but not by `qte_add_cxx_flags`, but rather its callee.

mwoehlke-kitware replied:
Hmm, yes, that looks plausible. Care to submit a patch to qtExtensions? (I wonder how no one else has noticed this...)

mwoehlke-kitware commented on Plugins/VvQueryService/KipQueryService/CMakeLists.txt:
```diff
@@ -43,6 +43,9 @@ vg_add_qt_plugin(${PROJECT_NAME}
 )
 
+set_property(TARGET ${PROJECT_NAME}
+  PROPERTY CXX_STANDARD 17)
```
Why is this needed?

neal-siekierski replied:
Because of the use of `[[maybe_unused]]` here:
https://github.com/neal-siekierski/vivia/blob/viame/dev/windows_fixes/Plugins/VvQueryService/KipQueryService/vvKipQuerySession.cxx#L462
Without it, VS complains that `/std:c++17` is required.

mwoehlke-kitware replied:
Sigh...

Removing the `[[maybe_unused]]` and adding a `Q_UNUSED(pluginsLoaded)` might be better. AFAIK this code is notionally supposed to work on GCC 4.8 which only "mostly" supports C++11 (although also some C++14 bits).

mwoehlke-kitware commented on Libraries/VtkVgCore/vtkVgPlotTimeline.cxx:
```diff
@@ -232,7 +232,7 @@ struct IdToPointXIter
     {}
 
-  double operator*()
+  double operator*() const
```
(FYI, it looks like this was also fixed by ace613f79b6a52f0404a1fa7e6cac105e602bba4.)

neal-siekierski commented:
I've switched to using `Q_UNUSED` instead of requiring C++17, and I've removed the qtextensions workaround in anticipation of its [`fix-interface-requirements`](https://github.com/Kitware/qtextensions/compare/fix-interface-requirements) branch getting merged.
